project_wrf.ipynb

```diff
@@ -0,0 +1,360 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "cc3f99ae-d46d-4172-8087-c7010f82c5f6",
+   "metadata": {},
+   "source": [
+    "# Projecting WRF data\n",
+    "\n",
+    "This notebook creates the coordinate reference system info for a single WRF file using the standard WRF geogrid file (`geo_em.d01.nc`) associated with the particular runs. The WRF data here are from the 1km runs done for the SERDP Fish and Fire project. This notebook draws heavily on [this blog post: \"Map projections in WRF: it's easy to get it wrong\" by Fabien Maussion](https://fabienmaussion.info/2018/01/06/wrf-projection/) for achieving the desired outcome - thank you Fabien!\n",
+    "\n",
+    "## Workflow:\n",
+    "\n",
+    "Load libraries and read WRF geogrid file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "cd52867a-10cc-4fd8-bcf5-23cff882f832",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import xarray as xr\n",
+    "from pyproj import Proj, Transformer\n",
+    "from pyproj.crs import CRS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "53f47881-28a1-483c-b876-a26a5816b985",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# SERDP WRF geogrid file\n",
+    "geo_ds = xr.open_dataset(\"/atlas_scratch/kmredilla/WRF/serdp/geo_em.d01.nc\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a34eb3a2-8af7-49a4-88ce-2ebaad99535b",
+   "metadata": {},
+   "source": [
+    "Create the projection info fro reprojecting lat/lon of WRF grid centerpoint to WRF CRS:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "fa35cf79-2912-43b3-98a7-23f257b58732",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create the WRF projection\n",
+    "wrf_proj = Proj(\n",
+    "    proj=\"lcc\", # projection type: Lambert Conformal Conic\n",
+    "    lat_1=geo_ds.TRUELAT1, lat_2=geo_ds.TRUELAT2, # Cone intersects with the sphere\n",
+    "    lat_0=geo_ds.MOAD_CEN_LAT, lon_0=geo_ds.STAND_LON, # Center point\n",
+    "    a=6370000, b=6370000 # This is it! The Earth is a perfect sphere\n",
```

Comment:
great use of comments!! seriously love it. Once you start intersecting cones and spheres...you are earning your geo-dollars that's for sure

Reply:
Hah honestly these comments are from the blog.. but super helpful here for sure!